src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+
+/**
+ * This class deals with making sense of the user command.
+ * @author Sim Jun Heng
+ * @version CS2103T AY21/22 Sem 2
+ */
+public class Parser {
+    private String command;
+    private String desc = null;
+    private String date = null;
+
+    /**
+     * Creates a parser object after parsing user inputs.
+     *
+     * @param str the string entered by the user.
+     */
+    public Parser(String str) {
+        String[] temp = str.split(" ");
+        command = temp[0];
+        if (temp.length > 1) {
+            int index = 1;
+            while (index < temp.length) {
+                if (!temp[index].equals("/by") && !temp[index].equals("/at")) {
+                    if (desc == null) {
```

Comment:
An alternative approach could be to use `str.indexOf(...)` to search for "/by" and "/at" and then use `str.substring(...)` to help extract the different sections of the input string. I think it could help save on the while loops and also help improve the readability of your code. What do you think?